src/java.base/share/classes/sun/security/ssl/SessionTicketExtension.java

```diff
@@ -403,11 +404,14 @@ public byte[] produce(ConnectionContext context,
             chc.statelessResumption = true;
 
             // If resumption is not in progress, return an empty value
-            if (!chc.isResumption || chc.resumingSession == null) {
+            if (!chc.isResumption || chc.resumingSession == null
+                    || chc.resumingSession.getPskIdentity() == null
+                    || !Arrays.asList(ProtocolVersion.PROTOCOLS_10_12)
+                        .contains(chc.resumingSession.getProtocolVersion())) {
```

Comment:
would  `chc.resumingSession.getProtocolVersion().useTLS13PlusSpec()` read better for your last condition ? Might save on Array allocations also ?

Reply:
good point!